x/concentrated-liquidity/store_test.go

```diff
@@ -75,3 +77,77 @@ func (s *KeeperTestSuite) TestGetAllPositionsWithVaryingFreezeTimes() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestParseFullPositionFromBytes() {
+	defaultFrozenUntil := s.Ctx.BlockTime().Add(DefaultFreezeDuration)
+	defaultAddress := s.TestAccs[0]
+	cdc := s.App.AppCodec()
+
+	tests := map[string]struct {
+		key          []byte
+		val          []byte
+		expectingErr bool
+	}{
+		"Empty val": {
+			key:          types.KeyFullPosition(defaultPoolId, defaultAddress, DefaultLowerTick, DefaultUpperTick, defaultFrozenUntil),
+			val:          []byte{},
+			expectingErr: true,
+		},
+		"Empty key": {
+			key:          []byte{},
+			val:          cdc.MustMarshal(&model.Position{Liquidity: DefaultLiquidityAmt, FrozenUntil: defaultFrozenUntil}),
+			expectingErr: true,
+		},
+		"Random key": {
+			key:          []byte{112, 12, 14, 4, 5},
+			val:          cdc.MustMarshal(&model.Position{Liquidity: DefaultLiquidityAmt, FrozenUntil: defaultFrozenUntil}),
+			expectingErr: true,
+		},
+		"Using not full key (wrong key)": {
+			key:          types.KeyPosition(defaultPoolId, defaultAddress, DefaultLowerTick, DefaultUpperTick),
+			val:          cdc.MustMarshal(&model.Position{Liquidity: DefaultLiquidityAmt, FrozenUntil: defaultFrozenUntil}),
+			expectingErr: true,
+		},
+		"One key separator missing in key": {
+			key:          []byte(fmt.Sprintf("%s%s%s%d%s%d%s%d", []byte{0x02}, defaultAddress, "|", defaultPoolId, "|", DefaultLowerTick, "|", DefaultUpperTick)),
+			val:          cdc.MustMarshal(&model.Position{Liquidity: DefaultLiquidityAmt, FrozenUntil: defaultFrozenUntil}),
+			expectingErr: true,
+		},
+		"Wrong position prefix": {
+			key:          []byte(fmt.Sprintf("%s%s%s%s%d%s%d%s%d", []byte{0x01}, "|", defaultAddress, "|", defaultPoolId, "|", DefaultLowerTick, "|", DefaultUpperTick)),
```

Comment:
Can we replicate this for each of pool id, lower tick, upper tick, and frozen until, please?

This is to ensure full coverage since we attempt to parse each of those fields and have an error check for it.

Reply:
hello @p0mvn. Changes addressed